org.opent2t.sample.temperaturesensor.superpopular/com.wink.thermostat.temperaturesensor/js/thingTranslator.js

```diff
@@ -1,100 +1,103 @@
-
-        if (typeof this.device != 'undefined') {
-            if (typeof (this.device.props) !== 'undefined') {
-                var props = JSON.parse(this.device.props);
-
-                if (typeof (props.access_token) !== 'undefined') {
-                    accessToken = props.access_token;
-                } else {
-                    console.log('props.access_token is undefined.');
-                }
-
-                if (typeof (props.id) !== 'undefined') {
-                    deviceId = props.id;
-                } else {
-                    console.log('props.id is undefined.');
-                }
-            } else {
-                console.log('props is undefined.');
-            }
-        } else {
-            console.log('device is undefined.');
+
+    get(state) {
+        if (this.desired_state[state] !== undefined) {
+            return this.desired_state[state];
         }
+        else {
+            return this.last_reading[state];
+        }
+    }
+}
 
-        console.log('javascript initialized.');
-        logDeviceState(this.device);
-    },
-
-    disconnect: function () {
-        console.log('disconnect called.');
-        logDeviceState(this.device);
-    },
-
-    getCurrentTemperature: function (callback) {
-        console.log('getCurrentTemperature called.');
-
-        var options = {
-            'protocol': 'https:',
-            'host': 'api.wink.com',
-            'path': '/thermostats/' + deviceId,
-            'headers': {
-                'authorization': 'Bearer ' + accessToken,
-                'Accept': 'application/json'
-            },
-            'method': 'GET'
-        };
-
-        var req = https.get(options, function (res) {
-            var body = '';
-            res.setEncoding('utf8');
-            res.on('data', function (data) {
-                body += data;
-            });
+// Helper method to convert the device schema to the translator schema.
+function deviceSchemaToTranslatorSchema(deviceSchema) {
 
-            res.on('end', function () {
-                if (res.statusCode != 200) {
-                    callback('error: ' + res.statusCode);
-                } else {
-                    var data = JSON.parse(body).data;
+    var stateReader = new StateReader(deviceSchema.desired_state, deviceSchema.last_reading);
 
-                    if (!!data.last_reading && !!data.last_reading.temperature) {
-                        callback(data.last_reading.temperature);
-                    }
-                }
-            });
+    var temperatureUnits = stateReader.get('units').temperature;
+
+    var result = {
+        id: deviceSchema['object_type'] + '.' + deviceSchema['object_id'],
+        n: deviceSchema['name'],
+        rt: 'org.opent2t.sample.temperaturesensor.superpopular',
+        ambientTemperature: { temperature: stateReader.get('temperature'), units: temperatureUnits }
+    };
+
+    return result;
+}
+
+var deviceId;
+var deviceType = 'thermostats';
```

Comment:
So Wink thermostats map to both thermostats and temperaturesensors?

Could this cause any problems? Which do we return when enumerating devices?